engine/execution/state/state.go

```diff
@@ -224,23 +224,34 @@ func (s *state) NewStorageSnapshot(
 
 type RegisterUpdatesHolder interface {
 	UpdatedRegisters() flow.RegisterEntries
+	UpdatedRegisterSet() map[flow.RegisterID]flow.RegisterValue
```

Comment:
The function below uses both of these. I think that is unnecessary. We could only have `UpdatedRegisterSet() map[flow.RegisterID]flow.RegisterValue` (which means we could just call it `UpdatedRegisters() map[flow.RegisterID]flow.RegisterValue`)

Reply:
Yeah, however it requires the caller to convert into a slice. And we probably don't want the caller to decide the order, the order of updates should be deterministic, that's `UpdatedRegisters` sort the slice internally before returning.

What about returning both map and slice, since both of them are used below?
It's possible, but I'm just afraid it's a bit confusing, returning the same data in two forms in one API. That's why I created a separate API for returning different form of the same data. 

Thoughts?

Reply:
`RegisterUpdatesHolder` is only used here and the only use of `UpdatedRegisters` is then converted into a map anyway, so order is not conserved.

I'm just not sure the function below (`CommitDelta`) needs to use both these interface methods.

Reply:
No, the order is actually conserved by `	keys, values := RegisterEntriesToKeysValues(updatedRegisters)`

The keys and values are ordered, which then be used to create `trieUpdate` by 
```
	newState, trieUpdate, err := ldg.Set(update)
```

The trieUpdate contains the ordered list of paths and payloads, the order has to be deterministic, otherwise it will produce different execution data id in the execution result.